pandas/core/apply.py

```diff
@@ -1309,6 +1353,50 @@ def apply(self):
     def transform(self):
         raise NotImplementedError
```

Comment:
This is superfluous now?

Reply:
Not exactly. We currently have this implemented on the base class `Apply`. It should not be called for e.g. GroupBy. Without this, if it accidentally gets called it would generate a perhaps confusing error; this is more straightforward to understand. Also - users should never encounter this; only developers.

That said, this is definitely a code smell. I think it may be good to move transform into `core.apply`.